internal/faults/faults.go

```diff
@@ -48,9 +48,9 @@ var (
 // protocol-specific header lookup. Using an interface here rather than a
 // function type avoids any potential closure requirements of a function.
 type Headers interface {
-	// Lookup returns the value of a protocol-specific header with the
-	// given key.
-	Lookup(ctx context.Context, key string) (string, error)
+	// LookupValues returns the values of a protocol-specific header with
+	// the given key.
+	LookupValues(ctx context.Context, key string) ([]string, error)
```

Comment:
the problem with this change is that thrift THeader does not support 1:N mapping from header to value.

since we are already doing a lot of text manipulation in the header value anyways, just use a single value, and combine with `;` or something if we need multiple there?

Reply:
As mentioned offline, only HTTP takes advantage of this, and the number of times we expect there to be multiple HTTP header values on separate lines is extremely low anyways. Thrift and GRPC just package a single header up in a slice and return it.

Multiple values for fault injection are supported via the RFC 9110 comma-separated header value syntax. i.e. `foo: bar; baz, quux` represents the header `foo` having 2 values: `bar; baz` and `quux`. The semicolon is used as the single header value field delimiter, while the comma is used as the delimiter between multiple separate header values.